### ISSUE TITLE
Fix `no-duplicate-at-import-rules` false negatives for imports with `supports` and `layer` conditions

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#6997]
### What minimal example or steps are needed to reproduce the bug?

```css
@import url('foo.css') layer supports(display: grid) tv, screen;
@import url('foo.css') layer supports(display: grid) screen;
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-standard"
  ]
}
```

### How did you run Stylelint?

https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBLAtgBwgJwC4AIBXbAGwAoByAMwggDowBnRigSnxIEMBPGbfRoQxY8jMlBSMMXbvHwBzbCijtcANwA0AsNhgwAdgG4AOvtSYcBYuWq0GzNhx58BQkbjESpMuYuXtGHT0jEA0QKhQSGAA5TjQ4RBgADzjpGHtGUPAIfQj5BBBgU3x8YxBk3AMoRjK5AG1iktKQRlxuKJIUfVwAWkhclHke1s59KE5sKDLGgF1TAF8s-ryAMRw0TlwCgCtGHKzYDEzEIv0Sstb2mE7u2uauStayjUaLto6u3uXB4dxR8cmdzKDxgTxACxA8yAA

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}
```

### What did you expect to happen?

I expected an error :

```
Unexpected duplicate @import rule foo.css [(no-duplicate-at-import-rules)](https://stylelint.io/user-guide/rules/no-duplicate-at-import-rules)[2:1-60]
```

### What actually happened?

No error was reported

### Do you have a proposal to fix the bug?

I propose that we consider `layer` and `supports` conditions to be part of the cache key.

see : https://github.com/stylelint/stylelint/pull/6996/files#diff-6e755a443d64a1885d2445c671076e00a6a54ebd01eb205b97d7dd21fe358e63R79-R141

**Comment by romainmenke** [comment]
When working on https://github.com/stylelint/stylelint/issues/6848 I found that `no-duplicate-at-import-rules` does not support the latest syntax.

https://www.w3.org/TR/css-cascade-5/#at-ruledef-import

```
@import [ <url> | <string> ]
        [ layer | layer(<layer-name>) ]?
        [ supports( [ <supports-condition> | <declaration> ] ) ]?
        <media-query-list>? ;
```

I suggest that we continue to split the list of media queries but add `layer` and `supports` conditions as a prefix to each individual media query.

```css
@import url("a.css") layer(base) supports(display: grid) screen, tv;
```

has keys :
- `layer(base) supports(display: grid) screen`
- `layer(base) supports(display: grid) tv`

**Comment by jeddy3** [comment]
Thank you for the detailed write-up. The proposal SGTM!
